src/hotspot/share/cds/aotClassInitializer.cpp

```diff
@@ -34,11 +35,18 @@ bool AOTClassInitializer::can_archive_initialized_mirror(InstanceKlass* ik) {
     return false;
   }
 
+  if (ik->is_hidden()) {
+    return HeapShared::is_archivable_hidden_klass(ik);
+  }
+
   if (ik->is_initialized() && ik->java_super() == vmClasses::Enum_klass()) {
     return true;
-  } else if (ik->name()->equals("jdk/internal/constant/PrimitiveClassDescImpl") ||
-             ik->name()->equals("jdk/internal/constant/ReferenceClassDescImpl") ||
-             ik->name()->equals("java/lang/constant/ConstantDescs")) {
+  }
+
+  Symbol* name = ik->name();
+  if (name->equals("jdk/internal/constant/PrimitiveClassDescImpl") ||
```

Comment:
Could we / should we start using `vmSymbols` constants for these? I think you can directly compare the `Symbol*`-s then, like `name == vmSymbols::jdk_internal_constant_PrimitiveClassDescImpl_Holder() `.

Reply:
Not every class in this table is inside vmSymbols, and I don't want to have to edit vmSymbols whenever a new class is added here (and having to do garbage collection in vmSymbols when a class is removed from here).

I pushed a new version that's more table driven, and avoid most strcmp calls by filtering with the symbol length.